components/org.wso2.carbon.identity.outbound.provisioning.connector.office365/src/main/java/org/wso2/carbon/identity/outbound/provisioning/connector/office365/Office365ProvisioningConnector.java

```diff
@@ -177,10 +178,57 @@ protected String createUser(ProvisioningEntity provisioningEntity) throws Identi
+     *
+     * @param provisioningEntity user to be provisioned
+     * @throws IdentityProvisioningException if the user can not be created in the Azure AD
+     */
+    protected void updateUser(ProvisioningEntity provisioningEntity) throws IdentityProvisioningException {
+
+        String provisionedUserId = provisioningEntity.getIdentifier().getIdentifier();
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            JSONObject user = buildUserAsJson(provisioningEntity);
+            String updateUserEndpoint = Office365ConnectorConstants.OFFICE365_USER_ENDPOINT + '/' + provisionedUserId;
+            HttpPatch patchRequest = new HttpPatch(updateUserEndpoint);
+            setAuthorizationHeader(patchRequest);
+
+            StringEntity requestBody = new StringEntity(user.toString());
+            requestBody.setContentType(Office365ConnectorConstants.CONTENT_TYPE_APPLICATION_JSON);
+            patchRequest.setEntity(requestBody);
+            patchRequest.setHeader(Office365ConnectorConstants.CONTENT_TYPE, Office365ConnectorConstants
+                    .CONTENT_TYPE_APPLICATION_JSON);
+
+            try (CloseableHttpResponse response = httpclient.execute(patchRequest)) {
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_NO_CONTENT) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully updated the provisioned user with id " + provisionedUserId + " in " +
+                                "the Azure Active Directory.");
+                    }
+                } else {
+                    JSONObject jsonResponse = new JSONObject(new JSONTokener(new InputStreamReader(
+                            response.getEntity().getContent())));
+                    String errorMessage = jsonResponse.getJSONObject("error").getString("message");
+
+                    log.error("Received response status code: " + response.getStatusLine().getStatusCode() + " "
+                            + response.getStatusLine().getReasonPhrase() + " with the message '" + errorMessage +
+                            "' while updating the user with id " + provisionedUserId + " in the Azure Active " +
+                            "Directory.");
 
-        log.warn("Update user is not implemented.");
-        // TODO: 8/14/18 Implement update user logic
+                    if (log.isDebugEnabled()) {
+                        log.debug("The response received from server : " + jsonResponse.toString());
+                    }
+                }
+            } catch (IOException | JSONException e) {
+                throw new IdentityProvisioningException("Error while executing the update operation in user " +
+                        "provisioning.", e);
+            }
+        } catch (IOException e) {
+            log.error("Error while closing HttpClient.");
```

Comment:
Shouldn't we throw the error? Anyhow the original exception need to be recorded (at least as a debug log)

Reply:
Updated to print the stack trace of the exceptions